common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationResultFactory.java

```diff
@@ -0,0 +1,95 @@
+package com.microsoft.identity.common.internal.providers.microsoft.microsoftsts;
+
+import android.content.Intent;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftAuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResultFactory;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationStatus;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.util.HashMap;
+
+public class MicrosoftStsAuthorizationResultFactory extends AuthorizationResultFactory {
+
+    private static final String TAG = MicrosoftStsAuthorizationResultFactory.class.getSimpleName();
+
+    public static final String MSSTS_AUTHORIZATION_FINAL_URL = "com.microsoft.identity.client.finalUrl";
+
+    @Override
+    public AuthorizationResult createAuthorizationResult(final int resultCode, final Intent data) {
```

Comment:
Another place where we could reduce the number of `return` statements for easier logging, debuggability, whatever

Reply:
Done.